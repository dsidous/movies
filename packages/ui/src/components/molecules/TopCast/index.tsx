import React from 'react';
import { Link } from 'react-router-dom';

import { Cast } from '@atoms/index';

import useStyles from './styles';
import { Cast_Credit } from '@typesRoots/movie';

interface Props {
  cast: Cast_Credit[];
  fullLink: string;
}

const TopCast: React.FC<Props> = ({ cast, fullLink }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.castWrapper}>
        {cast.map((person) => (
          <Cast cast={person} type='top' />
        ))}
      </div>

      <Link className={classes.fullCaCrBtn} to={fullLink}>
        Full Cast & Crew
      </Link>
    </div>
  );
};

export default TopCast;
