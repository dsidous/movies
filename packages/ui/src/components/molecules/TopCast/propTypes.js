import { array } from 'prop-types';

export const propTypes = {
  cast: array,
};

export const defaultProps = {
  cast: [],
};
