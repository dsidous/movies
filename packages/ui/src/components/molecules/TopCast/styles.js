import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      gridArea: 'mcast',

      [theme.breakpoints.up('sm')]: {
        margin: '0 auto',
      },

      '& h4': {
        padding: '20px 0',
      },
    },

    castWrapper: {
      display: 'grid',
      gridGap: 10,
      gridTemplateColumns: 'repeat(auto-fit, minmax(7em, 1fr))',

      [theme.breakpoints.up('sm')]: {
        gridGap: 20,
        gridTemplateColumns: 'repeat(6, 1fr)',
      },
    },

    fullCaCrBtn: {
      background: 'none',
      border: 'none',
      cursor: 'pointer',
      fontSize: '1.2em',
      fontWeight: 'bold',
      margin: '30px 0 0',

      '&:hover': {
        opacity: '.6',
      },
    },
  }),
  { name: 'topCast' },
);
