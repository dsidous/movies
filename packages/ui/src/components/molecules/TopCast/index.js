import React from 'react';

import { propTypes, defaultProps } from './propTypes';
import Cast from '../../atoms/Cast';

const TopCast = ({ cast, handleFullCrewClick }) => (
  <div className="cast">
    <h4>Top Billed Cast</h4>
    <div className="cast-wrapper">
      {cast.map(person => (
        <Cast cast={person} type="top" key={person.id} />
      ))}
    </div>

    <button
      type="button"
      className="full-crew-btn"
      onClick={handleFullCrewClick}
    >
      Full Cast & Crew
    </button>
  </div>
);

TopCast.propTypes = propTypes;
TopCast.defaultProps = defaultProps;

export default TopCast;
