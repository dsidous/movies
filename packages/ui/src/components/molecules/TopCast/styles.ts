import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      gridArea: 'mcast',
    },

    castWrapper: {
      display: 'grid',
      gridGap: 10,
      gridTemplateColumns: 'repeat(auto-fit, minmax(7em, 1fr))',

      [theme.breakpoints.up('sm')]: {
        gridGap: 20,
        gridTemplateColumns: 'repeat(6, 1fr)',
      },
    },

    fullCaCrBtn: {
      display: 'block',
      fontSize: '1.2em',
      fontWeight: theme.fontWeightBold,
      marginTop: theme.spacing(4),
    },
  }),
  { name: 'topCast' },
);
