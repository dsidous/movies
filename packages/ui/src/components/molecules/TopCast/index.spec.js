import React from 'react';
import { shallow } from 'enzyme';
import TopCast from '.';

const mockProps = {
  cast: [
    {
      name: 'test',
      id: 1,
      character: 'test',
      profile_path: 'path/to/image',
    },
    {
      name: 'test',
      id: 1,
      character: 'test',
      profile_path: 'path/to/image',
    },
  ],
  handleFullCrewClick: jest.fn(),
};

describe('Molecules/TopCast', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<TopCast {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
