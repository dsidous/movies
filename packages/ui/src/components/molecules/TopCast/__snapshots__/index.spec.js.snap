// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/TopCast should render as expected 1`] = `ShallowWrapper {}`;
